packages/gatsby-cli/src/structured-errors/construct-error.ts

```diff
@@ -1,25 +1,37 @@
 import stackTrace from "stack-trace"
 import { errorSchema } from "./error-schema"
-import { errorMap, defaultError, IErrorMapEntry } from "./error-map"
+import { defaultError, ErrorId, errorMap, IErrorMapEntry } from "./error-map"
 import { sanitizeStructuredStackTrace } from "../reporter/errors"
 import { IConstructError, IStructuredError } from "./types"
 // Merge partial error details with information from the errorMap
 // Validate the constructed object against an error schema
-const constructError = ({
-  details: { id, ...otherDetails },
-}: IConstructError): IStructuredError => {
-  const result: IErrorMapEntry = (id && errorMap[id]) || defaultError
+const constructError = (
+  { details: { id, ...otherDetails } }: IConstructError,
+  suppliedErrorMap: Record<ErrorId, IErrorMapEntry>
+): IStructuredError => {
+  let errorMapEntry = defaultError
+
+  if (id) {
+    // Look at original errorMap, ids cannot be overwritten
+    if (errorMap[id]) {
+      errorMapEntry = errorMap[id]
+    }
+
+    if (suppliedErrorMap[id]) {
+      errorMapEntry = suppliedErrorMap[id]
+    }
+  }
```

Comment:
It's probably my mistake in a previous comment but it should be
```suggestion
  if (id) {
    // Look at original errorMap, ids cannot be overwritten
    if (errorMap[id]) {
      errorMapEntry = errorMap[id]
    } else if (suppliedErrorMap[id]) {
      errorMapEntry = suppliedErrorMap[id]
    }
  }
```